extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1837,5 +1840,40 @@ describes.realWin(
         ).to.be.false;
       });
     });
+
+    describe('page loading', () => {
+      it('should load the active page on layout', async () => {
+        const pages = await createStoryWithPages(
+          2,
+          ['page-1', 'page-2'],
+          false
+        );
+        env.sandbox.stub(story, 'mutateElement').callsFake((mutator) => {
+          mutator();
+          return Promise.resolve();
+        });
+        await story.buildCallback();
+        await story.layoutCallback();
+        expect(pages[0].hasAttribute('distance')).to.be.true;
+      });
+
+      it('should load the inactive pages after the active page is loaded', async () => {
+        const pages = await createStoryWithPages(
+          2,
+          ['page-1', 'page-2'],
+          false
+        );
+        env.sandbox.stub(story, 'mutateElement').callsFake((mutator) => {
+          mutator();
+          return Promise.resolve();
+        });
+        await story.buildCallback();
+        await story.layoutCallback();
+        expect(pages[1].hasAttribute('distance')).to.be.false;
+        pages[0].dispatchEvent(new CustomEvent(EventType.PAGE_LOADED));
+        await nextTick();
+        expect(pages[1].hasAttribute('distance')).to.be.true;
```

Comment:
How would we know if we break the feature and load page2 before the active page is loaded? These tests would pass even without your changes

Reply:
The other tests can help with that, but I also added a check so this new test fails if page 2 loads before page 1